### ISSUE TITLE
Crash when doing a long press in the content of a webview

### ISSUE THREAD

**Issue by eriksquare** [phrase/phrase-android#31]
Phrase version 3.0.5, AppCompat version 1.3.0.

Steps to reproduce:
- set `minifyEnabled true`
- open a webview and try to copy-paste something or just select some test
- app crashes when the contextual selection menu shows up

to workaround this i've found in my tests that we can:
- set `minifyEnable false`
- comment the `Phrase.getDelegate(this, super.getDelegate())` line in my code

Also, this issue has been found in our live app in multiple phones and android versions. 

EDIT- also, i've used app bundles, i don't know if that is meaningful or not

**Comment by DarkMinstrel** [comment]
Hi Erik! Thanks for your report. 
Could you please share the stack trace?

**Comment by eriksquare** [comment]
```
    --------- beginning of crash
2021-06-22 09:59:14.546 6219-6219/com.myapp.debug E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.myapp.debug, PID: 6219
    android.content.res.Resources$NotFoundException: Resource ID #0x20c0016
        at android.content.res.ResourcesImpl.getValue(ResourcesImpl.java:216)
        at android.content.res.Resources.getInteger(Resources.java:1094)
        at org.chromium.ui.base.DeviceFormFactor.b(chromium-SystemWebViewGoogle.aab-stable-447211403:4)
        at org.chromium.content.browser.selection.SelectionPopupControllerImpl.j(chromium-SystemWebViewGoogle.aab-stable-447211403:1)
        at u8.onCreateActionMode(chromium-SystemWebViewGoogle.aab-stable-447211403:8)
        at Fv.onCreateActionMode(chromium-SystemWebViewGoogle.aab-stable-447211403:2)
        at com.android.internal.policy.DecorView$ActionModeCallback2Wrapper.onCreateActionMode(DecorView.java:2396)
        at com.android.internal.policy.DecorView.startActionMode(DecorView.java:899)
        at com.android.internal.policy.DecorView.startActionModeForChild(DecorView.java:855)
        at android.view.ViewGroup.startActionModeForChild(ViewGroup.java:985)
        at android.view.ViewGroup.startActionModeForChild(ViewGroup.java:985)
        at android.view.ViewGroup.startActionModeForChild(ViewGroup.java:985)
        at android.view.ViewGroup.startActionModeForChild(ViewGroup.java:985)
        at android.view.ViewGroup.startActionModeForChild(ViewGroup.java:985)
        at android.view.ViewGroup.startActionModeForChild(ViewGroup.java:985)
        at android.view.ViewGroup.startActionModeForChild(ViewGroup.java:985)
        at android.view.ViewGroup.startActionModeForChild(ViewGroup.java:985)
        at android.view.View.startActionMode(View.java:6811)
        at Im.a(chromium-SystemWebViewGoogle.aab-stable-447211403:1)
        at org.chromium.content.browser.selection.SelectionPopupControllerImpl.J(chromium-SystemWebViewGoogle.aab-stable-447211403:10)
        at ke0.a(chromium-SystemWebViewGoogle.aab-stable-447211403:10)
        at lh0.i(chromium-SystemWebViewGoogle.aab-stable-447211403:4)
        at g6.run(chromium-SystemWebViewGoogle.aab-stable-447211403:3)
        at android.os.Handler.handleCallback(Handler.java:873)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:280)
        at android.app.ActivityThread.main(ActivityThread.java:6706)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)

```

this is a standard apk (form runconfiguration) with `minifyEnabled true` (so app bundles are not part of the problem)
By keeping the `minifyEnabled true` and commenting the delegate everything works.
When googling i've found a similar issue in another library (which i actually don't use) https://github.com/B3nedikt/restring/issues/42 maybe there is a common cause.

**Comment by DarkMinstrel** [comment]
Unfortunately, we couldn't reproduce the crash. 
It would be very helpful if you provide a minimal example project that demonstrates the issue.

**Comment by angelolloqui** [comment]
We are suffereing the same issue. I believe it is not related to OTA but to the usage of ViewPump or some interceptors for wrapping the context. Some related issues:

https://github.com/InflationX/ViewPump/issues/60
https://github.com/InflationX/ViewPump/issues/64

**Comment by eriksquare** [comment]
> @eriksquare were you using ViewPump or some interceptors?

never used ViewPump, and you mean view interceptors? i guess only emoji compat could be something related...

**Comment by RogerParis** [comment]
Is there any chance that an invalid id is passed in into getText? There is also a possibility that this is related to an issue with webviews on certain Android devices/versions. Are you using webviews that might be related to this?

**Comment by RogerParis** [comment]
@eriksquare another client had a similar issue and this is what helped him track down the problem:

"Maybe the following way helps you.
We checked all available source IDs in both our app and a Phrase site. We found that one ID is existing for the Thai or Vietnamese language, but not for the default (English). Which was the source of the crash and error."

Maybe this feedback can help you.

**Comment by Savrov** [comment]
The same here. And like the author said, it crashes cause of 
`Phrase.getDelegate(this, super.getDelegate())`

**Comment by Krillsson** [comment]
Just to chime in. We are also seeing this crash. Any update on the troubleshooting? @DarkMinstrel @theSoenke?

**Comment by theSoenke** [comment]
Hi @Krillsson, the investigation took a bit longer than expected but the plan is now to release a solution this month. I'll post an update here once there is some news

**Comment by theSoenke** [comment]
Hi, this issue should have been resolved in version 3.3.1 https://github.com/phrase/phrase-android/releases/tag/3.3.1
